src/SchemaRegistry.ts

```diff
@@ -117,10 +122,18 @@ export default class SchemaRegistry {
 
     const schema = await this.getSchema(registryId)
 
-    return encode(schema, registryId, jsonPayload)
+    const serdes = serdesTypeFromSchemaType(schema.type)
+    let serializedPayload
+    try {
+      serializedPayload = serdes.serialize(schema, payload, serdesOpts)
+    } catch (error) {
+      throw new ConfluentSchemaRegistrySerdesError(error)
+    }
+
+    return encode(registryId, serializedPayload)
   }
 
-  public async decode(buffer: Buffer): Promise<any> {
+  public async decode(buffer: Buffer, serdesOpts?: {}): Promise<any> {
```

Comment:
Another option would be to have it be:

```ts
interface SerdesOptions {
  [SchemaType.AVRO]?: AvroSerdesOptions
  [SchemaType.JSON]?: JsonSerdesOptions
  [SchemaType.PROTO]?: ProtoSerdesOptions
}
```

That way the interface remains pretty simple while still allowing you to pass in options for both Protobuf and Avro, in case you are consuming both.